### PR TITLE
Extinguisher cleanup

### PR DESCRIPTION
**Pull request by Cyberboss**
Shitcode has to die

### CODE REVIEW

**Comment by AnturK** [comment]
Thanks, have a 🌟

**Comment by octareenroon91** [comment]
Could you make extinguishers work with sinks in this PR?

**Comment by xxalpha** [comment]
I looked at the changes and immediately regretted my decision.

File: code/game/objects/items/weapons/extinguisher.dm [hunk 1]
```diff
@@ -145,31 +127,25 @@
...
+			W.create_reagents(5)
+			reagents.trans_to(W, 1)
+			new /datum/forced_movement(W, my_target, 0.5, TRUE, CALLBACK(GLOBAL_PROC, /proc/WaterStepCallback))
 	else
 		return ..()
 
+/proc/WaterStepCallback(datum/forced_movement/FM)
+	var/obj/effect/particle_effect/water/W = FM.victim
+	var/turf/T = get_turf(W)
+	W.reagents.reaction(T)
```
Comment (octareenroon91) [inline code/game/objects/items/weapons/extinguisher.dm]: `method = TOUCH` is currently default for `reaction()` but you may care to set it specifically.

**Review by duncathan — APPROVED** [review]

**Comment by Jalleo** [comment]
Lets hope we can make this any reagent after the freeze.

**Comment by octareenroon91** [comment]
@Jalleo That's more or less been the case. There's just no way to load anything but water, and from water tanks.

File: code/game/objects/items/weapons/extinguisher.dm [hunk 2]
```diff
@@ -113,28 +112,11 @@
...
-				sleep(2)
-				step(B, movementdirection)
-				sleep(2)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
```
Comment (PKPenguin321) [inline code/game/objects/items/weapons/extinguisher.dm]: the fuck was this accomplishing

**Comment by Cyberboss** [comment]
DNM yet, accidentally gave them infinite range

File: code/game/objects/items/weapons/extinguisher.dm (hunk 2 above)
Comment (RemieRichards) [inline code/game/objects/items/weapons/extinguisher.dm]: @PKPenguin321 this is extinguisher chair pushing, it moves you, sleeps, moves you, sleeps, moves you... etc.

**Comment by lzimann** [comment]
fix the thing pls

**Comment by fludd12** [comment]
Not to add work to your plate, but notably, the water spray from the extinguisher applies it's reagents several times. You may want to fix this if you want to try @Jalleo's suggestion of filling extinguishers with any reagent.

For reference, in Paracode, at least, hitting someone full-blast xenobio execution chamber style with the equivalent of sulfuric acid does 450 damage.

**Comment by Cyberboss** [comment]
TOO LAZY

**Comment by octareenroon91** [comment]
what is going on now?

**Pull request closed without merging**